controllers/clusterversion_controller.go

```diff
@@ -124,47 +122,181 @@ func (c *ClusterVersionReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
-	if err := c.ensureConsolePlugin(); err != nil {
-		c.log.Error(err, "unable to deploy client console")
+	c.log.Info("Loading input resources")
+
+	// Load the cluster version
+	c.clusterVersion.Name = req.Name
+	c.clusterVersion.Namespace = req.Namespace
+	if err := c.get(&c.clusterVersion); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Load the console plugin
+	c.consoleDeployment.Name = console.DeploymentName
+	c.consoleDeployment.Namespace = c.OperatorNamespace
+	if err := c.get(&c.consoleDeployment); err != nil {
+		c.log.Error(err, "failed to get the deployment for the console")
+		return ctrl.Result{}, err
+	}
+
+	// Load the operator configmap
+	c.operatorConfigMap.Name = operatorConfigMapName
+	c.operatorConfigMap.Namespace = c.OperatorNamespace
+	if err := c.get(&c.operatorConfigMap); err != nil && !k8serrors.IsNotFound(err) {
 		return ctrl.Result{}, err
 	}
 
-	instance := configv1.ClusterVersion{}
-	if err = c.Client.Get(context.TODO(), req.NamespacedName, &instance); err != nil {
+	c.log.Info("Starting reconciliation")
+	if result, err := c.reconcilPhases(); err == nil {
+		c.log.Error(err, "An error was encountered during reconciliation")
 		return ctrl.Result{}, err
+	} else {
+		c.log.Info("Reconciliation completed successfully")
+		return result, nil
 	}
+}
+
+func (c *ClusterVersionReconciler) reconcilPhases() (ctrl.Result, error) {
 
-	if err := csi.InitializeSidecars(c.log, instance.Status.Desired.Version); err != nil {
+	// Reconcile console related resources
+	if err := c.reconcileNginxConfigMap(); err != nil {
+		return ctrl.Result{}, err
+	}
+	if err := c.reconcileConsoleService(); err != nil {
+		return ctrl.Result{}, err
+	}
+	if err := c.reconcileConsolePlugin(); err != nil {
+		c.log.Error(err, "unable to deploy client console")
+		return ctrl.Result{}, err
+	}
+
+	// Reconcile CSI related resources
+	if err := csi.InitializeSidecars(c.log, c.clusterVersion.Status.Desired.Version); err != nil {
 		c.log.Error(err, "unable to initialize sidecars")
 		return ctrl.Result{}, err
 	}
+	if err := c.reconcileSecurityContextConstraints(); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := c.reconcileMonConfigMap(); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := c.reconcileEncConfigMap(); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := c.reconcileCephFSDeployment(); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := c.reconcileCephFSDaemonSet(); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := c.reconcileRBDDeployment(); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := c.reconcileRBDDaemonSet(); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := c.reconcileCephFSCSIDrivers(); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := c.reconcileRBDCSIDriver(); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := c.reconcilePrometheusRule(); err != nil {
+		return reconcile.Result{}, err
```

Comment:
to be consistent, we can use either `ctrl.Result` (existing) or `reconcile.Result` throughout, I suggest the former and the only reason is `ctrl.Result` is a higher level alias.

Reply:
done